inferno-vc/test/Spec.hs

```diff
@@ -4,222 +4,45 @@
-            _ -> expectationFailure ""
-          -- h2 is an autosave so it's fine
-          runOperation vcClientEnv (deleteAutosavedFunction h2) $ \() -> pure ()
-
-    it "history" $ do
-      o1 <- createObj Init
-      runOperation vcClientEnv (pushFunction o1) $ \h1 -> do
-        o2 <- createObj $ MarkedBreakingWithPred h1
-        runOperation vcClientEnv (pushFunction o2) $ \h2 -> do
-          o3 <- createObj $ MarkedBreakingWithPred h2
-          runOperation vcClientEnv (pushFunction o3) $ \h3 -> do
-            o4 <- createObj $ MarkedBreakingWithPred h3
-            runOperation vcClientEnv (pushFunction o4) $ \h4 -> do
-              runOperation vcClientEnv (fetchVCObjectHistory h4) $ \metas ->
-                (map obj metas) `shouldBe` [h4, h3, h2, h1]
-
-    it "history of clone" $ do
-      o1 <- createObj Init
-      runOperation vcClientEnv (pushFunction o1) $ \h1 -> do
-        o2 <- createObj $ MarkedBreakingWithPred h1
-        runOperation vcClientEnv (pushFunction o2) $ \h2 -> do
-          o3 <- createObj $ CloneOf h2
-          runOperation vcClientEnv (pushFunction o3) $ \h3 -> do
-            o4 <- createObj $ MarkedBreakingWithPred h3
-            runOperation vcClientEnv (pushFunction o4) $ \h4 -> do
-              runOperation vcClientEnv (fetchVCObjectHistory h4) $ \metas ->
-                (map obj metas) `shouldBe` [h4, h3, h2]
-
-    it "history of clone of clone" $ do
-      o1 <- createObj Init
-      runOperation vcClientEnv (pushFunction o1) $ \h1 -> do
-        o2 <- createObj $ CloneOf h1
-        runOperation vcClientEnv (pushFunction o2) $ \h2 -> do
-          o3 <- createObj $ CloneOf h2
-          runOperation vcClientEnv (pushFunction o3) $ \h3 -> do
-            o4 <- createObj $ MarkedBreakingWithPred h3
-            runOperation vcClientEnv (pushFunction o4) $ \h4 -> do
-              runOperation vcClientEnv (fetchVCObjectHistory h4) $ \metas ->
-                (map obj metas) `shouldBe` [h4, h3, h2]
 
 main :: IO ()
 main =
   withSystemTempDirectory "vc_store_" $ \vcPath -> do
     putStrLn $ "Store is at: " ++ (show vcPath)
 
     putStr "Starting Inferno VC..."
+    let txtTracer = IOTracer $ contramap T.unpack simpleStdOutTracer
     _ <-
       forkIO $
-        runServerConfig
-          (Proxy :: Proxy Int)
-          (Proxy :: Proxy Int)
-          ServerConfig
-            { _serverHost = "127.0.0.1",
-              _serverPort = 13077,
-              _vcPath = vcPath
-            }
-    man <- newManager defaultManagerSettings
-    let vcClientEnv =
-          mkVCClientEnv man $
-            BaseUrl
-              { baseUrlScheme = Http,
-                baseUrlHost = "127.0.0.1",
-                baseUrlPort = 13077,
-                baseUrlPath = []
+        FSOps.withEnv vcPath txtTracer $ \env ->
+          runServerConfig
+            (Proxy :: Proxy Int)
+            (Proxy :: Proxy Int)
+            txtTracer
+            (flip FSOps.runInfernoVCFilesystemM env)
```

Comment:
Yeah. The spec is defined in `Inferno.VersionControl.Testing` so can be imported from client code and test a backend without touching inferno-vc

Reply:
Hmm sorry I don’t understand, isn’t this file inside the `inferno-vc` library package?

Reply:
Of course, if inferno-vc grows another "builtin" backend it can be tested here too

Reply:
> Hmm sorry I don’t understand, isn’t this file inside the `inferno-vc` library package?

Sorry, I'll try to reword better: I moved it to the library so it can be reused outside inferno-vc (eg: by plow-inferno to test cardith backend)

Reply:
Ah yes, makes sense thanks!